TACC/Core-CMS
# Update to Django 3.2 (for TUP CMS)

## Overview

Update Django to v3.2 and bump deprecated dependencies, for [tup-cms](https://github.com/TACC/tup-ui/tree/main/apps/tup-cms).

## Related

- [TUP-453](https://jira.tacc.utexas.edu/browse/TUP-453)
- mimics https://github.com/TACC/Core-CMS/pull/625 but off of `dev/tup-cms`
- includes https://github.com/TACC/Core-CMS/pull/631
- requires https://github.com/TACC/Core-CMS-Resources/pull/176
- required by https://github.com/TACC/tup-ui/pull/225

## Changes

- update dependencies
- rename `…-cms` to `…_cms` (e.g. `example-cms` to `example_cms`)
- rename `…-org` to `…_org` (e.g. `texascale-org` to `texascale_org`)

## Testing

0. [Build Core-CMS image.](https://confluence.tacc.utexas.edu/pages/viewpage.action?pageId=276664110#HowToBuild&DeployCamino/Jenkins/WatchtowerCMSProject-Core-CMS)
    - build: https://jenkins01.tacc.utexas.edu/job/Core_CMS_Build/1053/console
2. **Either** spin up [TUP-CMS](https://github.com/TACC/tup-ui/tree/main/apps/tup-cms) locally.
    **Or** [deploy TUP CMS](https://confluence.tacc.utexas.edu/pages/viewpage.action?pageId=276664110#HowToBuild&DeployCamino/Jenkins/WatchtowerCMSProject-TUP-CMS) on **dev** server.
    - deploy: https://jenkins01.tacc.utexas.edu/view/TUP/job/tup-cms-deploy/19/console
3. Verify CMS allows add/publish/delete pages.
4. Verify assets are loaded and pages are unchanged from before deploy.
5. Follow test steps on https://github.com/TACC/tup-ui/pull/225.

## UI

See https://github.com/TACC/tup-ui/pull/225.

## Notes

#625 requires to many updates (to Core-CMS-Resources), and we want to update TUP ASAP, so this branch and PR is made.

### Known Issues

See https://github.com/TACC/tup-ui/pull/225.
